packages/class/src/DataEventSource.js

```diff
@@ -0,0 +1,70 @@
+// @ts-check
+
+export default class DataEventSource {
+  /**
+   * @param {string} url
+   */
+  constructor(url) {
+    this.url = url;
+    /** @type {Function[]} */
+    this.handlers = [];
+    this.eventSource = null;
+  }
+
+  /**
+   * @private
+   */
+  start() {
+    if (this.eventSource) {
+      return;
+    }
+
+    this.eventSource = new EventSource(this.url, {
+      withCredentials: true,
+    });
+    this.eventSource.addEventListener('message', this.handleEvent);
+  }
+
+  /**
+   * @private
+   */
+  stop() {
+    if (!this.eventSource) {
+      return;
+    }
+
+    this.eventSource.removeEventListener('message', this.handleEvent);
+    this.eventSource.close();
+    this.eventSource = null;
+  }
+
+  /**
+   * @private
+   * @param {MessageEvent} data
+   */
+  handleEvent = data => {
```

Comment:
Why arrow function?

Reply:
need store context when call
`this.eventSource.addEventListener('message', this.handleEvent);`
and then
`this.eventSource.removeEventListener('message', this.handleEvent);`

Reply:
Not obvious. Can't using "just method", without arrow?

Reply:
nope, can't using without arrow,
another solution is create handler in constructor with bind for pass the same handler to eventSource.
i prefer arrow solution